### DISCUSSION TITLE
Run SQL query from JavaScript

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9122] [Help] [answered]
Hi there, potentially a stupid question, but can I run a query from JavaScript?

**Comment by melohagan** [comment, marked as answer]
Hey @ecxs 

The JavaScript is sandboxed, so you cannot make API calls or run queries. There is the **Execute Query** action that can be used, for example: https://docs.budibase.com/docs/custom-queries#tutorial-posting-form-data-to-a-custom-query

Otherwise you can build a custom component and do an API call from within it, but that requires some dev work.

**Reply by ecxs** [reply]
Thanks @melohagan ... for my use case (setting a computer name/id), I first build a automation that triggered on row create, then searched for data in the last record, and finally updated the row. I then figure out, that this also could be achieved more directly by calling an Execute Query Action from the Save button when creating a new record ...

![Screenshot 2022-12-22 at 08 25 44](https://user-images.githubusercontent.com/4005405/209080823-338d096b-0a4d-4438-847e-be91286fc630.png)
